src/persist-types/src/codec_impls.rs

```diff
@@ -629,10 +629,6 @@ impl ColumnRef<()> for BooleanBuffer {
             .as_any()
             .downcast_ref::<BooleanArray>()
             .ok_or_else(|| format!("expected BooleanArray but was {:?}", array.data_type()))?;
-        if array.logical_nulls().is_some() {
-            return Err("unexpected validity for non-optional bool".to_owned());
-        }
```

Comment:
I had to remove this because when we encode `SourceData(Err(...))` we push a `Null` into a bool column. So even if the `ColumnType` specifies `nullable: false` it's still possible for the inner column to contain `null` values.

Reply:
The impl of `ColumnRef` for `BooleanBuffer` is the only impl that has an assert like this. I don't know for sure but my guess is that when roundtripping through Parquet the boolean column gets the validity buffer of the higher level `StructArray` which is how this assert ends up failing